nexus-writer/src/main.rs

```diff
@@ -253,6 +265,13 @@ fn process_frame_assembled_event_list_message(nexus_engine: &mut NexusEngine, pa
     match root_as_frame_assembled_event_list_message(payload) {
         Ok(data) => match nexus_engine.process_event_list(&data) {
             Ok(run) => {
+                let metadata_result: Result<FrameMetadata, _> = data.metadata().try_into();
```

Comment:
```suggestion
                match data.metadata().try_into::<FrameMetadata>();
```

Reply:
`try_into()` doesn't take generics, so it would have to be either
```rust
match TryInto::<FrameMetadata>::try_into(data.metadata()) {
```
or
```rust
match data.metadata().try_into() as Result<FrameMetadata,_> {
````
I think the second would be better, but not sure which one you'd prefer as standard.